packages/admin/admin/src/common/ReadOnlyAdornment.tsx

```diff
@@ -0,0 +1,61 @@
+import { Lock } from "@comet/admin-icons";
+import { type ComponentsOverrides, css, InputAdornment, type InputAdornmentProps, type Theme, useThemeProps } from "@mui/material";
+
+import { createComponentSlot } from "../helpers/createComponentSlot";
+import { type ThemedComponentBaseProps } from "../helpers/ThemedComponentBaseProps";
+
+export type ReadOnlyAdornmentClassKey = "root";
+
+export type ReadOnlyAdornmentProps = ThemedComponentBaseProps<{
+    root: typeof InputAdornment;
+}> & {
+    position?: InputAdornmentProps["position"];
+    inputIsReadOnly: boolean;
+} & Omit<InputAdornmentProps, "position">;
+
+export const ReadOnlyAdornment = (inProps: ReadOnlyAdornmentProps) => {
+    const {
+        position = "end",
+        children = <Lock fontSize="inherit" />,
+        inputIsReadOnly,
+        slotProps,
+        ...restProps
+    } = useThemeProps({
+        props: inProps,
+        name: "CometAdminReadOnlyAdornment",
+    });
+
+    if (!inputIsReadOnly) {
+        return null;
+    }
```

Comment:
I find it strange that a component called `ReadOnlyAdornment` has a required `inputIsReadOnly` prop and if it's false it renders nothing. IMO the ReadOnlyAdornment should only be rendered if it's in a read only input anyway, so

```
{readOnly && <ReadOnlyAdornment />}
```

instead of 

```
<ReadOnlyAdornment inputIsReadOnly={readOnly} />
```